Source/SwiftLintBuiltInRules/Rules/Style/PreferSelfInStaticReferencesRuleExamples.swift

```diff
@@ -72,6 +72,9 @@ enum PreferSelfInStaticReferencesRuleExamples {
                 let c: C = C()
                 func f(c: C) -> KeyPath<C, Int> { \\Self.i }
             }
+        """, excludeFromDocumentation: true),
+        Example("""
+            class C: A<C.i> {}
```

Comment:
We like to have the examples as valid/compilable Swift code. This example isn't valid because `C.i` doesn't exist.

Also there are a few other cases I'd like to see test cases for. Your example, having it as

```swift
class C1<T> {}
class C2: C1<C2.C3> { // rule not triggering here
    class C3 {}
}
```

is one case. Another one is just

```swift
class C1<T> {}
class C2: C1<C2> {} // rule not triggering here
```

without the inner type.

Now, your fix also causes new issues in some [OSS projects](https://github.com/realm/SwiftLint/pull/5084#issuecomment-1608180503) which we have to check and make sure the findings and fixes are okay. There are at least two other cases we need to have tests for. These are:

```swift
class S1<T> {
    class S2 {}
    func f() {
        let s1 = S1<S1.S2>() // rule not triggering here
        let s2 = S1<S1>()    // rule could trigger here but shouldn't
    }
```

This applies to classes only. In structs, the rule may trigger in both cases.

Reply:
Added two more examples and edited the one that was added.

Sorry for doing a force-push, now you can't do "review changes since your last review". Just wanted to fix it in one commit.